### PR TITLE
Added Cost explorer nightly

### PR DESCRIPTION
**Pull request by athiruma**

### CODE REVIEW

File: cloud_governance/common/aws/cost_explorer/cost_explorer_operations.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    """
+    This class extracts the price data from AWS Cost Explorer
+    """
+
+    def __init__(self):
+        self.cost_explorer_client = boto3.client('ce')
+
+    def get_cost_usage(self, tag: str, metrics_type: str):
+        """
+        This method extracts the IAM price by Tag provided
```
Comment (ebattat) [inline cloud_governance/common/aws/cost_explorer/cost_explorer_operations.py:17]: This method extracts the IAM price by Tag provided - is it the right term  IAM price ?

File: cloud_governance/cost_expenditure/generate_cost_explorer_report.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    def __get_daily_cost_by_tags(self):
+        """
+        This method extracts the costs by tags and upload to elastic search
+        @return:
+        """
+        if not self.metric_type:
+            self.metric_type = 'BlendedCost'
+        data_house = {}
+        tags = ['User', 'Budget', 'Project', 'Manager', 'Owner', 'LaunchTime', 'Name', 'Email']
```
Comment (ebattat) [inline cloud_governance/cost_expenditure/generate_cost_explorer_report.py:47]: Can we pass it as  environment variables ?

File: cloud_governance/cost_expenditure/generate_cost_explorer_report.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                data.append({tag: name, 'Cost': round(float(amount), 3)})
+        return data
+
+    def __get_daily_cost_by_tags(self):
+        """
+        This method extracts the costs by tags and upload to elastic search
+        @return:
+        """
+        if not self.metric_type:
+            self.metric_type = 'BlendedCost'
```
Comment (ebattat) [inline cloud_governance/cost_expenditure/generate_cost_explorer_report.py:45]: Can you get it from environment variable ?

File: cloud_governance/cost_expenditure/generate_cost_explorer_report.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+from multiprocessing import Process
+
+from cloud_governance.common.aws.cost_explorer.cost_explorer_operations import CostExplorerOperations
+from cloud_governance.common.logger.init_logger import logger
+from cloud_governance.common.elasticsearch.elasticsearch_operations import ElasticSearchOperations
+
+
+class GenerateCostExplorerReport:
+
+    def __init__(self, es_host: str, es_port: str, es_index: str, metric_type: str):
```
Comment (ebattat) [inline cloud_governance/cost_expenditure/generate_cost_explorer_report.py:11]: Can someone run cost explorer and extract the data to file and not to ES ?

File: cloud_governance/common/aws/cost_explorer/cost_explorer_operations.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        @return:
+        """
+        today = datetime.now()
+        start_time = today - timedelta(1)
```
[the remark this answers is not in the document]
Comment (athiruma) [reply cloud_governance/common/aws/cost_explorer/cost_explorer_operations.py:21]: yes. 
In cost explorer, i filter today. It shows NoTagKey.
We only got one day back data.

File: cloud_governance/cost_expenditure/generate_cost_explorer_report.py (hunk 2 above)
Comment (athiruma) [reply cloud_governance/cost_expenditure/generate_cost_explorer_report.py:47]: yes we can.

File: cloud_governance/cost_expenditure/generate_cost_explorer_report.py (hunk 3 above)
Comment (athiruma) [reply cloud_governance/cost_expenditure/generate_cost_explorer_report.py:45]: yes, if they didn't provide. By default we will take BlendedCost

File: cloud_governance/cost_expenditure/generate_cost_explorer_report.py (hunk 4 above)
Comment (athiruma) [reply cloud_governance/cost_expenditure/generate_cost_explorer_report.py:11]: yes. By giving full file_path.

Comment (ebattat) [reply cloud_governance/cost_expenditure/generate_cost_explorer_report.py:11]: So please change the es fields to be empty by default 
es_host: str= '', es_port: str= '', es_index: str='', metric_type: str='BlendedCost'

File: cloud_governance/common/aws/cost_explorer/cost_explorer_operations.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ebattat) [reply cloud_governance/common/aws/cost_explorer/cost_explorer_operations.py:21]: The method name should be:
get_daily_cost_usage

**Review by ebattat — APPROVED** [review]
/LGTM

**Pull request merged**